library/src/cache/patch_manager.rs

```diff
@@ -70,7 +70,7 @@ pub trait ManagePatches {
 
     /// Records that the patch with number patch_number booted successfully and is
     /// safe to use for future boots.
-    fn record_boot_success_for_patch(&mut self, patch_number: usize) -> Result<()>;
+    fn record_boot_success(&mut self) -> Result<()>;
```

Comment:
That means this is moving from being a "stateless" to "stateful" api, which may be fine.  But probably as part of that you'd want the patch_manager to always be created with a specific current_patch?  And then never allow that to be changed?

Otherwise you might have the problem of not easily being able to intuit from the call sites as to what "current_patch" they're affecting?